### ISSUE TITLE
Sagemaker Compatibility with Detectron2

### ISSUE THREAD

**Issue by AliButtar** [aws-samples/amazon-sagemaker-pytorch-detectron2#8]
Hi,

First, thanks for the great code, It is really easy to navigate through and understand and make changes.

So the problem is I have been trying out various experiments to run detectron2 on sagemaker but for some reason, any other version than the one which is being used does not run. I tried to use torch version 1.8 and 1.9 from Deep learning containers and made appropriate changes to the Dockerfile and also changed the link to the detectron2 to try out versions 0.5 and 0.6 but the training never starts. 

I also made changes in the `training.py` file to not use the custom trainer and the data you had but instead use the default Trainer, Evaluator and also load up data from COCO instances to make it a lot more generalized. I have been successful in running this training on the current detecton2 version and Pytorch version but for some reason when I try to upgrade to a newer one, it gives out strange errors and just fails the training job. 

Can you please guide me on what could be the problem here? Is there any code inside `training.py` that is specific to detectron2 version 0.4 hence it only runs on that? I can provide any additional information that you require to help me debug this.

I also tried to run this repo as it is and didn't change anything except PyTorch versions and detectron2 version but it still failed.

Thanks,
AliButtar

**Comment by paoloirrera** [comment]
Hello,

Sorry for replying with large delay.

I did not test with versions past 0.4 so I can't really tell. However, since you say "I tried to use torch version 1.8 and 1.9 from Deep learning containers" I assume that you're using [SageMaker Pytorch DL containers](https://github.com/aws/deep-learning-containers/blob/master/available_images.md). These containers compile PyTorch version that are optimized to run on SageMaker. However, Detectron2, according to [installation guide](https://detectron2.readthedocs.io/en/latest/tutorials/install.html), requires PyTorch and torchvision versions that come from the official Torch channel. This is why the Dockerfile contains a line:

`RUN pip install --upgrade torch==1.6.0+cu101 torchvision==0.7.0+cu101 -f https://download.pytorch.org/whl/torch_stable.html`

You will have to to update torch and torchvision versions according to the D2 version that you want to use.

Moreover, it would be helpful to share the log error from the SageMaker training job. D2 versions past 0.4 may introduce changes that are not compatible with the 0.4 version.

**Comment by AliButtar** [comment]
Hi,

No problem.

I did update those lines accordingly to match with detectron2 dependencies and I tried various versions but none of them seemed to work instead of detectron2 version 0.4. I will try to attach logs later but It was always failed at different parts when trying out different versions. Sometimes it would go right up until training was about start and then it would fail. 

Thanks,
Ali Buttar
